### PR TITLE
Updated zlib to version 1.2.12

### PR DESCRIPTION
**Pull request by tedjpoole**

### CODE REVIEW

**Comment by phlax** [comment]
@tedjpoole could you fix DCO please

**Review by keith — DISMISSED** [review]

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -484,15 +484,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "zlib",
         project_desc = "zlib compression library",
         project_url = "https://zlib.net",
-        version = "79baebe50e4d6b73ae1f8b603f0ef41300110aa3",
-        # Use the dev branch of zlib to resolve fuzz bugs and out of bound
-        # errors resulting in crashes in zlib 1.2.11.
-        # TODO(asraa): Remove when zlib > 1.2.11 is released.
-        sha256 = "155a8f8c1a753fb05b16a1b0cc0a0a9f61a78e245f9e0da483d13043b3bcbf2e",
+        version = "21767c654d31d2dccdde4330529775c6c5fd5389",
```
Comment (jwendell) [inline bazel/repository_locations.bzl:487]: is it better to use the tag `v1.2.12` instead of the SHA?

Comment (phlax) [reply bazel/repository_locations.bzl:487]: yep - release version/tags are definitely preferable

**Comment by phlax** [comment]
copying out CI

```console

DependencyChecker ERROR ERRORS Summary [release_dates]:
--------------------------------------------------------------------------------
Mismatch: net_zlib 2022-03-28 != 2022-03-27

```

**Comment by phlax** [comment]
(UTC date mismatch)

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (tedjpoole) [reply bazel/repository_locations.bzl]: Done

**Comment by tedjpoole** [comment]
Corrected zlib release date

**Review by phlax — APPROVED** [review]
lgtm, thanks @tedjpoole

**Pull request merged**